### PR TITLE
Fix stratosphere-dist to include the scala jars

### PR DESCRIPTION
**Pull request by aljoscha**
Also fix an annoying bug that made tests fail because a dependency on
scala-reflect was wrongly in the "test" scope and also add an example of
how to create a PlanAssembler for a scala job in WordCount

### CODE REVIEW

**Comment by rmetzger** [comment]
Looks good

File: pact-scala/pact-scala-examples/src/main/scala/eu/stratosphere/scala/examples/wordcount/WordCount.scala [hunk 1]
```diff
@@ -39,6 +40,12 @@ object RunWordCount {
   }
 }
 
+class WordCountDescriptor extends PlanAssembler {
+  override def getPlan(args: String*) = {
+    new WordCount().getPlan(args(0), args(1))
```
Comment (StephanEwen) [inline pact-scala/pact-scala-examples/src/main/scala/eu/stratosphere/scala/examples/wordcount/WordCount.scala]: Does WordCount actually execute with a DOP > 1? It seems that the Args are parsed, but a specified DOP option is never used.

File: stratosphere-dist/src/main/assemblies/bin.xml [hunk 2]
```diff
@@ -138,6 +138,19 @@
...
+		<fileSet>
+			<!-- copy jar files of pact job examples -->
+			<directory>../pact-scala/pact-scala-examples/target</directory>
+			<outputDirectory>examples/pact-scala/</outputDirectory>
+			<fileMode>0644</fileMode>
+			<includes>
+				<include>*.jar</include>
+			</includes>
+			<excludes>
+				<exclude>pact-examples-${project.version}.jar</exclude>
```
Comment (StephanEwen) [inline stratosphere-dist/src/main/assemblies/bin.xml]: Does that exclude the wrong project jar file? Seems it should be called "pact-scala-examples-${project.version}.jar"

**Comment by aljoscha** [comment]
Right as always. :D I'll add the correct exclude and add DOP as a parameter.

**Comment by aljoscha** [comment]
I think this badboy should be ready for merge now. Care to have another look?

**Pull request merged**